poppy/optics.py

```diff
@@ -81,14 +83,15 @@ def __str__(self):
     def getPhasor(self, wave):
         raise NotImplementedError("getPhasor must be supplied by a derived subclass")
 
-    def sample(self, wavelength=2e-6, npix=512, grid_size=None, what='amplitude',
+    @quantity_input(wavelength=u.meter)
```

Comment:
I liked that `quantity_input` was namespaced under `utils.quantity_input` in `fresnel.py`. Might keep people from thinking it's the exact same as the astropy one... Other naming idea: `optional_units_input` or `optional_units_args`?